src/builtin/ws.rs

```diff
@@ -62,29 +63,29 @@ pub struct WebSocket {
 /// # Example
 ///
 /// ```
-/// # use via::{App, Request, Response, Next};
-/// # let mut app = App::new(());
-/// #
-/// use via::builtin::ws::Message;
+/// use via::{App, BoxError, Payload};
+/// use via::builtin::ws::{self, Message, WebSocket};
 ///
-/// app.at("/echo").respond(via::ws(async |mut socket, context| {
+/// #[tokio::main]
+/// async fn main() -> Result<(), BoxError> {
+///     let mut app = App::new(());
+///
+///     // GET /echo ~> web socket upgrade.
+///     app.at("/echo").respond(via::ws(echo));
+/// }
+///
+/// async fn echo(mut socket: WebSocket, _: ws::Context) -> via::Result<()> {
 ///     while let Some(message) = socket.next().await {
-///         match message {
-///             Message::Text(text) => {
-///                 socket.send(text).await?;
-///             }
-///             Message::Binary(binary) => {
-///                 socket.send(binary).await?;
-///             }
-///             Message::Close(_) => {
-///                 break;
-///             }
-///             _ => {}
+///         if matches!(&message, Message::Close(_)) {
```

Comment:
The payload trait makes working with websocket messages easy if you want to treat text and binary messages the same. It also can be an easy way to unwrap the reason from a close message.

I imagine that many apps with either branch on close or handle each message type individually. The introduction of the Payload trait allows users of Via to decide how specific they want to get. It's also implemented for `Bytes` and `ByteString`.